examples/llama2/llama2_generate.json

```diff
@@ -60,6 +60,32 @@
                     "joiner": "\n\n"
                 }
             }
+        },
+        {
+            "name": "prompt_transformer_dummy_data",
+            "type": "TransformersDummyDataContainer",
+            "load_dataset_config": {
+                "params": {
+                    "batch_size": 2,
```

Comment:
- Please make these params (batch_size, max_seq_len etc., ) for TransformerDummyDataContainer have proper default values and remove them from .json configs in examples.
- Please have two separate builtin containers (one for prompt and another for token) with appropriate default values for seq lens

Reply:
Updated.
Keep `generative` as false by default since it will skip `position_id` and `past_key_value`. Also, the `generative=true` is used only for llama2_generate.json.

Please let me know if you want to set `generative` as true by default.

Reply:
Why is llama2_generate.json is special regarding this? 
@shaahji

Reply:
The dataset was evaluation is different because the expected input shapes are different for running as a generative model vs. non-generative model. Ideally, the difference should be driven by change in data config only but the setup was using dataloader_func so just carry forward.

This will be swapped out by a near future PR to support only data configs for metric configuration.

Reply:
llama2 is a generative model so I'm not sure what is the difference between running as a generative model vs. non-generative model.

Reply:
BTW, this is a just a nit-pick that should not block this PR.